include/bout.hxx

```diff
@@ -52,8 +52,7 @@
 #include "where.hxx"
 #include "bout/mesh.hxx"
 #include "bout/solver.hxx"
-
-const BoutReal BOUT_VERSION = BOUT_VERSION_DOUBLE; ///< Version number
+#include "bout/version.hxx"
```

Comment:
Do we need to include `bout/version.hxx` here? Won't that result in as many files being rebuilt as when the macro was in `bout.hxx` directly?

Reply:
I thought that this header was only included in user code (and `bout++.cxx`), turns out I forgot it's also included in `physicsmodel.hxx` which is then included in `solver.hxx`, etc. 

We don't currently have a one-true-public-header that bundles everything needed for user code. I think we probably do want one, and this can then go in there.

I'll remove it for now.

Reply:
So, I initially had the commit hash in a separate file, `revision.hxx`, and then I merged it with `version.hxx`.

However, the version info changes much less frequently than the commit. Maybe it does make more sense to have them in separate files? Currently only `bout++.cxx` uses the commit, whereas `physicsmodel.cxx` and a couple of tests use the version number.

Reply:
I think that makes sense. Minimising the number of files to recompile after changing the commit hash would be nice :+1:

Reply:
Ok, down to just `bout++.cxx` that needs recompiling on commit. CMake seems to still also rebuild all the tests when I think they only need relinking. I suspect it might be because it might not be tracking header includes across targets, but I might be doing something wrong.